c10/cuda/impl/CUDAGuardImpl.h

```diff
@@ -40,6 +40,13 @@ struct CUDAGuardImpl final : public c10::impl::DeviceGuardImplInterface {
   }
   Device getDevice() const override {
     DeviceIndex device = 0;
+
+    // For supporting tracing cuda model on cuda-less environment,
+    // if device is not available, use cuda:0 by default
+    if (c10::cuda::device_count() == 0) {
+      return Device(DeviceType::CUDA, 0);
```

Comment:
I don't feel good about returning a device if there are no devices. I'd feel better about returning a device if (1) there are no devices and (2) FakeTensorMode is active, but it's a bit difficult to predict what is going on here

Why exactly did you need to stub out getDevice() ?

Reply:
Removed changes in this file. 

With the NoopDeviceGuard, this change is no longer needed.